drivers/soundwire/intel.c

```diff
@@ -752,13 +759,9 @@ static int intel_post_bank_switch(struct sdw_intel *sdw)
 	 *
 	 * So, set the SYNCGO bit only if CMDSYNC bit is set for any Master.
 	 */
-	if (!(sync_reg & SDW_SHIM_SYNC_CMDSYNC_MASK)) {
-		ret = 0;
-		goto unlock;
-	}
+	if (sdw_intel_sync_check_cmdsync_unlocked(sdw))
+		ret = sdw_intel_sync_go_unlocked(sdw);
 
-	ret = sdw_intel_sync_go_unlocked(sdw);
-unlock:
```

Comment:
would it better to move the check if (ret < 0) below also into the same  if (sdw_intel_sync_check_cmdsync_unlocked(sdw)) {} block? then you wont have to init ret also

Reply:
usually the console logs are moved outside of a mutex critical section, that's why I left it this way. But the code can be improved indeed - well the starting point wasn't that great in the first place.